### ISSUE TITLE
battle in builds with cmake very slow, but not with with autoconf builds

### ISSUE THREAD

**Issue by armijnhemel** [OpenXcom/OpenXcom#1292]
Whenever I am doing a build with cmake the battles are very slow even when player movement speed is set to maximum. I cannot see that the cmake build does a debug build. I tested with an autoconf build (from just before autoconf support was removed) and with those builds battle is very fast.

The binaries are slightly different: the cmake build links to libGLU.so.1 whereas the autoconf build does not. When doing a build with Makefile.simple the battles are very fast. When building using Makefile.simple libGLU.so.1 is not linked.

Builds are done on a fully updated Fedora 33, but I also experienced this on a Fedora 30 machine. Builds are done in clean source checkouts.

I am quite puzzled by this.

**Comment by SupSuper** [comment]
Might be the same issue as #1284 ?

**Comment by armijnhemel** [comment]
> Might be the same issue as #1284 ?

Possibly, that's why I mentioned it is not a debug build. I checked the cmake settings, but it is very well possible I missed something. What are the recommended instructions to run a cmake build (as it isn't described in the README)?

**Comment by uqs** [comment]
I'm sorry to hear this causes problems, can you please provide and try the following (under some pastebin or maybe upload the logfiles zipped or something):

- verbose cmake build:
```
cmake -B build-reldeb -DCMAKE_BUILD_TYPE=RelWithDebInfo .
cmake --build build-reldeb -v     // or cd into build-reldeb and make VERBOSE=1
```
- same for the autoconf build
- please download a linux build artefact from https://github.com/OpenXcom/OpenXcom/actions  (click on the latest good run, scroll down, there are 4 binaries, 2 linux, 1 win, 1 macos). see how fast they are.
- provide ldd(1) output on a good vs. a bad binary
- provide size(1) output for both
- provide nm(1) output for both

FWIW, on FreeBSD openxcom links against both libGL and libGLU

I'm looking at cmake/Modules/FindOpenGL.cmake right now and there's talk about GLVND and maybe you can try setting that in CMakeLists.txt?

As for building with cmake, it's documented in the Wiki but also in https://github.com/OpenXcom/OpenXcom/blob/master/README-OSX.md (which is more developer centric)

**Comment by uqs** [comment]
Erm, ignore all that ldd/size stuff, just tar up your complete build dir and provide that somewhere so I can download and have a look. It must be something that CMake "finds" and autoconf doesn't or vice versa.

**Comment by armijnhemel** [comment]
OK, will do that later today and e-mail a link to you.

**Comment by armijnhemel** [comment]
> OK, will do that later today and e-mail a link to you.

I have sent a download link to the e-mail address listed in the Git commit log.

**Comment by armijnhemel** [comment]
Found it. It turns out that by default a debug build is done. I fixed it by doing:

    cmake -B build -DCMAKE_BUILD_TYPE=Release

as the first step. Then everything works as expected. So I guess that the proper "fix" here would be a documentation fix :-)

**Comment by MeridianOXC** [comment]
Solved in #1300
